components/treeview/drag-drop.md

```diff
@@ -49,13 +82,212 @@ The `OnDrop` event provides an object of type `TreeViewDropEventArgs` to its eve
 | `DestinationIndex` | `string` | The index where the drop will happen in the second component. |
 | `DestinationComponentId` | `string` | The `Id` of the second component in which the drop will happen. |
 
+## OnDragEnd Event
+
+The `OnDragEnd` event fires when a drag operation is ends. The event is triggered after the `OnDrop` Event and unlike it, `OnDragEnd` will fire even if the new location is not a Telerik component.
+
+### Event Arguments
+
+The `OnDragEnd` event handler receives as an argument an object of type `TreeViewDragEndEventArgs` that contains:
```

Comment:
What does the event argument contain if the user drops somewhere outside the TreeView?

Reply:
If the user drops on a component that works with draggable like Grid/Scheduler/Treelist/Treeview then the event argument is largely the same as OnDrop's argument. Otherwise, the properties of the argument are empty/null.